mapie/classification.py

```diff
@@ -625,11 +658,315 @@ def _fit_and_predict_oof_model(
+            y_pred_proba,
+            y_pred_index_last,
+            axis=1
+        )
+
+        return y_pred_proba_cumsum, y_pred_index_last, y_pred_proba_last
+
+    def _update_size_and_lambda(
+        self,
+        best_sizes: NDArray,
+        alpha_np: NDArray,
+        y_ps: NDArray,
+        lambda_: Union[NDArray, float],
+        lambda_star: NDArray
+    ) -> Tuple[NDArray, NDArray]:
+        """Update the values of the optimal lambda if the
+        average size of the prediction sets decreases with
+        this new value of lambda.
+
+        Parameters
+        ----------
+        best_sizes : NDArray of shape (n_alphas, )
+            Smallest average prediciton set size before testing
+            for the new value of lambda_
+        alpha_np : NDArray of shape (n_alphas)
+            Level of confidences.
+        y_ps : NDArray of shape (n_samples, n_classes, n_alphas)
+            Prediction sets computed with the RAPS method and the
+            new value of lambda_
+        lambda_ : NDArray of shape (n_alphas, )
+            New value of lambda_star to test
+        lambda_star : NDArray of shape (n_alphas, )
+            Actual optimal lambda values for each alpha.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            Arrays of shape (n_alphas, ) and (n_alphas) which
+            respectively represents the update values of lambda_star
+            and the new best sizes.
+        """
+
+        sizes = [
+            classification_mean_width_score(
+                y_ps[:, :, i]
+            ) for i in range(len(alpha_np))
+        ]
+
+        sizes_improve = (sizes < best_sizes)
+        lambda_star = (
+            sizes_improve * lambda_ + (1 - sizes_improve) * lambda_star
+        )
+        best_sizes = sizes_improve * sizes + (1 - sizes_improve) * best_sizes
+
+        return lambda_star, best_sizes
+
+    def _find_lambda_star(
+        self, y_pred_proba_raps: NDArray,
+        alpha_np: NDArray, include_last_label: Union[bool, str, None],
+        k_star: NDArray
+    ) -> Union[NDArray, float]:
+        """Find the optimal value of lambda for each alpha.
+
+        Parameters
+        ----------
+        y_pred_proba_raps: NDArray of shape (n_samples, n_labels, n_alphas)
+            Predictions of the model repeated on the last axis as many times
+            as the number of alphas
+        alpha_np : NDArray of shape (n_alphas, )
+            Level of confidences.
+        include_last_label : bool
+            Whether or not to include last label in
+            prediction sets
+        k_star: NDArray of shape (n_alphas, )
+            Values of k for the regularization.
+
+        Returns
+        -------
+        ArrayLike of shape (n_alphas)
+            Optimal values of lambda.
+        """
+        lambda_star = np.zeros(len(alpha_np))
+        best_sizes = np.ones(len(alpha_np)) * 1e10
+
+        for lambda_ in [.001, .01, .1, .2, .5]:
```

Comment:
In MapieRegressor we often use

for __, lambda_ in enumerate([.001, .01, .1, .2, .5]):

Reply:
And maybe give precision to the choice of the values?

Reply:
there is no need to use enumerate if you do not use the indices

Reply:
done for the choice